### PR TITLE
Pass api_token to hipchat_url

### PR DESCRIPTION
**Pull request by davidboren**
I just barely started poking around here but I'm pretty sure we need to pass this bad boy into hipchat_url (because it pulls from hipchat_api_token here: https://github.com/robertzk/hipchat/blob/master/R/send.R#L89)...  401 statuses errrrryday, son.

### CODE REVIEW

**Comment by davidboren** [comment]
I had to actually change more entries because when we use the "determine_target" function here: https://github.com/robertzk/hipchat/blob/master/R/message.R#L76 we eventually, without a cache, end up calling refresh_room_cache without a token here: https://github.com/robertzk/hipchat/blob/master/R/room.R#L52

Which resets us to the original token.

**Comment by davidboren** [comment]
Uggh.  sanitize_room also calls hipchat_room_id and multiple exported functions (which do not have api_token as an argument) call sanitize_room.  I'd be happy to add the token arguments, but now we're getting into modifying exported stuff...  Also, wouldn't it make sense to only use hipchat_api_token() as the default argument for exported functions?  Although that would require test overhauls.  Nevermind...

File: R/user.R [hunk 1]
```diff
@@ -34,7 +34,6 @@ hipchat_create_user <- function(name, password, email, title, mention_name,
                  is_group_admin = is_group_admin, timezone = timezone, method = 'POST')
   if (!missing(title)) params$title <- title
   if (!missing(mention_name)) params$mention_name <- mention_name
-  browser()
```
Comment (robertzk) [inline R/user.R]: Woah. Thanks!

Comment (davidboren) [inline R/user.R]: Hehe I thought that was mine.  Just grepped and killed all of em

**Comment by robertzk** [comment]
Yes, `hipchat_api_token()` should only be a default for exported functions. I'll handle the rest of the exported calls to `sanitize_room`.

File: R/history.R [hunk 2]
```diff
@@ -29,8 +30,8 @@
 #'   # Full Hipchat API output for the latest 75 messages.
 #' }
 hipchat_history <- function(room_name_or_id, date = 'recent', timezone = 'UTC', start_index = 0,
-                            max_results = 100, reverse = TRUE, full = FALSE) {
-  room_id <- sanitize_room(room_name_or_id)
+                            max_results = 100, reverse = TRUE, full = FALSE, api_token = hipchat_api_token()) {
```
Comment (abelcastilloavant) [inline R/history.R]: 🎉

**Pull request merged**